python/shotgun_data/shotgun_data_retriever.py

```diff
@@ -891,17 +891,26 @@ def _get_thumbnail_path(url, bundle, directory_only=False):
             #   /bundle_cache_location/thumbs/C1C2/C3C4/rest_of_hash.*
             cache_matches = glob.glob("%s.*" % cache_base)
             if len(cache_matches):
+                if len(cache_matches) > 1:
+                    # There may be a legacy ".jpeg" file cached here. If so, remove it.
+                    old_cache_file = "%s.jpeg" % cache_base
+                    if old_cache_file in cache_matches:
+                        bundle.log_warning("Removing legacy '.jpeg' cache file : %s" % old_cache_file)
+                        cache_matches.remove(old_cache_file)
+                        os.remove(old_cache_file)
+
+                    # Now if more than one cache match is found, log a warning.
+                    if len(cache_matches) > 1:
+                        bundle.log_warning("More than one cached file found for url '%s':" % url)
+                        [bundle.log_warning("    %s" % cache_match) for cache_match in cache_matches]
+                        bundle.log_warning(
+                            "Using '%s'.  If this is incorrect, try clearing your local site cache." %
```

Comment:
If i am animator using zero config, this message will be very hard to understand! Again, all of these messages should probably be debug.